cm-gateway/gateway-api-install/manifests/gateway-api.yaml

```diff
@@ -0,0 +1,6906 @@
+#! This file comes from https://raw.githubusercontent.com/projectcontour/contour/v1.22.0/examples/gateway/00-crds.yaml
```

Comment:
This should come from the official Gateway API releases, not from Contour as eventually we'll be wanting to use the same CRDs for testing with other gateway controllers i.e https://github.com/kubernetes-sigs/gateway-api/tree/v0.4.1/config/crd/v1alpha2

Generally I have some doubts about using the latest Contour and Gateway API v0.5.0. There were big changes with Gateway API 0.5.0 and we are making changes in cert-manager to support it that will come out in the next release https://github.com/cert-manager/cert-manager/pull/5376. Those changes should be to do with using the beta CRDs mostly, but I think we would be safer to use the same versions as we've tested with cert-manager v1.9.1 at least for the Gateway API

Reply:
Thanks for your reminding, do we use contour [v1.20.1](https://github.com/cert-manager/cert-manager/commit/5f1dfa8c2ba8e05a1f285cdc8a1dcf7d944eba48) + gateway api `v0.4.1` in this pr?

Reply:
I think that would make sense- that way we can be sure it works 👍🏼 
The Gateway API is very experimental still, so there seem to be big changes between versions

Reply:
I was just thinking about it. It is better to use the `kubectl apply -k 'github.com/kubernetes-sigs/gateway-api/config/crd?ref=v0.4.1'` to install gateway api that is provided by [gateway-api document](https://gateway-api.sigs.k8s.io/v1alpha2/guides/getting-started/#install-standard-channel) and we also use it in [cert-manager e2e test](https://github.com/cert-manager/cert-manager/blob/6947696b12428a1f9e5823c2e69de84b8dd08c1c/devel/addon/gatewayapi/install.sh#L22). I think we don't need to copy them to the locality only for viewing the deploy status. If there are really errors, the whole deploy procession will be faill, we can debug it through output logs.

In future, if it is supported other stable install way, we can update the terraform process.

Reply:
I agree that this makes sense 👍🏼 

Less maintenance work which is good 👍🏼